lib/shared/utils.js

```diff
@@ -78,3 +78,62 @@ exports.percentageToFloat = (percentage) => {
 
   return percentage / exports.PERCENTAGE_MAXIMUM
 }
+
+/**
+ * @summary Memoize a function
+ * @function
+ * @private
+ *
+ * @description
+ * This workaround is needed to avoid AngularJS from getting
+ * caught in an infinite digest loop when using `ngRepeat`
+ * over a function that returns a mutable version of an
+ * ImmutableJS object.
+ *
+ * The problem is that every time you call `myImmutableObject.toJS()`
+ * you will get a new object, whose reference is different from
+ * the one you previously got, even if the data is exactly the same.
+ *
+ * @param {Function} func - function that returns an ImmutableJS list
+ * @param {Function} comparer - function to compare old and new args and state
+ * @returns {Function} memoized function
+ *
+ * @example
+ * const getList = () => {
+ *   return Store.getState().toJS().myList;
+ * };
+ *
+ * const memoizedFunction = memoize(getList, angular.equals);
+ */
+exports.memoize = (func, comparer) => {
```

Comment:
Given how specific this is to AngularJS and lists, I wonder if just `memoize` is a "too generic" function name? \*shrug\*